lib/Parser/Parse.cpp

```diff
@@ -2250,6 +2267,14 @@ void Parser::EnsureStackAvailable()
 
 void Parser::ThrowNewTargetSyntaxErrForGlobalScope()
 {
+    // If we are parsing a previously deferred function, we can skip throwing the SyntaxError for `new.target` at global scope.
+    // If we are at global scope, we would have thrown a SyntaxError when we did the Upfront parse pass and we would not have
+    // deferred the function in order to come back now and reparse it.
+    if (m_parseType == ParseType_Deferred)
```

Comment:
Is this just a shortcut, or does it fix some issue?

Reply:
This is from the 'defer parse lambda' branch which is where this change to remove the global 'this' binding was pulled out of. The change here avoids the need to figure out if the lambda is actually global at defer parse time by skipping this unneeded check.

Reply:
OK. Thanks.